### DISCUSSION TITLE
Integration does set the number.evcc_loadpointXX_limit_soc to 100 %

### DISCUSSION THREAD

**Discussion by lucode** [marq24/ha-evcc#25] [Q&A]
Not sure if this really is caused by the integration or by evcc itself. But is definitely not caused by my automation.
I have an automation which switches between the two setpoints for day an charging at night. But occasionally the entity sets itself to 100%, which is quite annoying. So far I couldn't detect any regularity beside it gets triggered at 14:20 every day.
The other ones are rather random.

![image](https://github.com/user-attachments/assets/2fd10735-3c1c-4dcb-b7b6-fcc22179a80c)

What as well puzzles me is the fact, that the slider remains at 90% even the number is set to another value, either by automation or the ghost setting.
![image](https://github.com/user-attachments/assets/7ecb8c02-e06c-4c22-ba03-fd8ce5238c98)

**Comment by marq24** [comment]
> What as well puzzles me is the fact, that the slider remains at 90% even the number is set to another value, either by automation or the ghost setting. 

wow! - this is indeed quite scarry - how do you change the value of the veh_target_soc in your automation code?

**Reply by lucode** [reply]
I use Node-Red, but did check it with HA Automation, same result. Set number.
![image](https://github.com/user-attachments/assets/c2665c14-e990-41f1-9c66-08c64173ce42)

```
action: number.set_value
metadata: {}
data:
  value: '60'
target:
  entity_id: number.evcc_carport_charger_limit_soc
```

**Comment by marq24** [comment]
actually when you use ha-external code, then I could imagine, that something in the caches and data containers something will get out of sync... if you want to manipulate HA-entities, then please use HA only... (might be there is a bug in the API that will be used by Node-Red?!

What you can try (as alternative) is directly to use the evcc - API to manipulate you data - then the Integration will pick up the updates - at least 5min later. Since this is the only thing that I can test/verify here - when data in evcc changes (e.g via GUI or API), that then the integration will pick up this change in a reasonable amount of time <5min.

**Reply by lucode** [reply]
The problem is not that I can't manipulate the data, but that the entity is changing the value, triggered by an unknown source. 
The non working slider synchronization with the value is a miner problem, but as it is the only slider of this Integration it might be also caused by a bug. Can you verify in your setup that if you change the value via an automation, that the slider updates the value too?

**Reply by marq24** [reply]
here is what I did:

1'st Browser - Firefox: I have opened the Entity Details showing the slider and the current value
2'nd Browser - Chromium: I have the 'developer' action Open and sending the `number.set_value`

as soon as the action is fired the slider is updated...

**Comment by marq24** [comment]
just something additional you might like to check, enable the debug log for the integration and check, if with every write operation you trigger from your NodeRed code you see a corresponding `write result` followed by a `going to read all data` (not `frequent data`) in the HA logs...

**Reply by lucode** [reply]
Nothing wrong visible from the logs, but I try to catch the source from the 100% change at 14:20 today.

**Comment by marq24** [comment]
cause I am curious - was there really something that could be fixed - or it's it more, that it's "none-explainable" ?!

**Reply by lucode** [reply]
I'm not quite sure, but the issue with the difference in the showed percentage in the slider and the actual value was a js issue. 
About the setting to 100 % I'm still not sure, but I guess it comes from the evcc "addon". After an restart of the "addon" for instance. As I can easily correct the value back to my desired value I see it as solved.
